### ISSUE TITLE
Code chunks not detected when sending lines and chunks in qmd/rmd

### ISSUE THREAD

**Issue by jamespeapen** [R-nvim/R.nvim#214]
Replicate of #190.

NVIM v0.10.0
R.nvim 54982fb
nvim-treesitter 1e86056

This affects both Rmd and Quarto documents.

Using this minimal config produces the error:

`minimal.vim`:

```vim
call plug#begin()
Plug 'R-nvim/R.nvim'
Plug 'nvim-treesitter/nvim-treesitter'
call plug#end()

lua <<EOF
```
```lua

require'nvim-treesitter.configs'.setup {
    ensure_installed = {
        "lua",
        "markdown",
        "markdown_inline",
        "python",
        "r",
        "rnoweb",
        "yaml",
    }
}
```
```vim
EOF
```

with this document

````md
# Test

```{r}
iris
ls()
```

````

Some commands that work on a variable or function:
- `RSendParagraph`
- `RSendParagraphDown`
- `RSendChunkFH`
- `RNLeftPart`
- `RNRightPart`
- `RObjectPR`
- `RSummary`
- `RShowArgs`

Commands that don't work:

- `RSendLine`
- `RDSendLine`
- `RSendChunk`
- `RDSendChunk`

gives `Not inside R or python code chunk`. 

When adding `lualine_x = { require("r.utils").get_lang },` to my lualine config, it shows the insides of code-chunks as "markdown" instead of r, unless I add a blank line to the beginning of the chunk - which is when the commands above do work.

I've tried resetting my treesitter installation as mentioned in #190, but it didn't help.

Everything was working a few weeks ago and I haven't changed my configs so I'm not sure why it stopped. I tried doing a git bisect on the repo but couldn't find a past commit where it worked again so could it be caused by a treesitter/neovim update? The code chunks start being recognized when I run `:InspectTree`.

**Comment by jalvesaq** [comment]
I can't replicate the issue. Did you try to delete the `nvim-treesitter` directory to force the reinstallation of parsers?

```
cd ~/.local/share/nvim/lazy
rm -rf nvim-treesitter/
```

**Comment by jamespeapen** [comment]
I realized that it was because I had markdown highlighting disabled in my treesitter config:

```lua
  highlight = {
    enable = true,              -- false will disable the whole extension
    disable = { "markdown" },  -- list of language that will be disabled
    additional_vim_regex_highlighting = false,
  },
```

I thought it only disabled the highlights but retained all the other functionality, but apparently highlighting is necessary for detecting filetype?

I disabled highlighting because I didn't like that it concealed the code chunk language:

![image](https://github.com/user-attachments/assets/97124d69-3bab-43b7-9849-8df3e16a650f)

highlighting disabled:
![image](https://github.com/user-attachments/assets/edf62c08-8a6a-4979-bb59-56ad09bb23da)

**Comment by jalvesaq** [comment]
With tree-sitter, I prefer to put the chunk options as comments (`#| `) in the R code:

````md

```{r}
#| code-line-numbers: "1-2"

iris
ls()
```
````

I have this `~/.config/nvim/after/queries/markdown/highlights.scm`:

```
; extends

; Conceal curly braces in code block delimiters
(fenced_code_block 
  (fenced_code_block_delimiter)
  (info_string) @conceal
  (#set! conceal ""))

; Highlight include command
((paragraph) @paragraph
  (#match? @paragraph "^\\{\\{\\< ")
  ) @include @nospell
```

And this in my lazy config:

```lua
    -- Background of headers and code blocks in Quarto documents
    {
        "lukas-reineke/headlines.nvim",
        dependencies = "nvim-treesitter/nvim-treesitter",
        config = function ()
            require("headlines").setup({
                quarto = {
                    --- Copied from headlines.nvim doc, replacing parse_query with query.parse
                    query = vim.treesitter.query.parse(
                        "markdown",
                        [[
                            (atx_heading [
                                (atx_h1_marker)
                                (atx_h2_marker)
                                (atx_h3_marker)
                                (atx_h4_marker)
                            ] @headline)
         
                            (thematic_break) @dash
         
                            (fenced_code_block) @codeblock
         
                            (block_quote_marker) @quote
                            (block_quote (paragraph (inline (block_continuation) @quote)))
                            (block_quote (paragraph (block_continuation) @quote))
                            (block_quote (block_continuation) @quote)
                        ]]
                    ),
                    treesitter_language = "markdown",
                    headline_highlights = { "Headline" },
                    bullet_highlights = {
                        "@text.title.1.marker.markdown",
                        "@text.title.2.marker.markdown",
                        "@text.title.3.marker.markdown",
                        "@text.title.4.marker.markdown",
                    },
                    bullets = { "◉", "○", "✸", "•" },
                    codeblock_highlight = "CodeBlock",
                    dash_highlight = "Dash",
                    dash_string = "-",
                    quote_highlight = "Quote",
                    quote_string = "┃",
                    fat_headlines = true,
                    fat_headline_upper_string = "▃",
                    fat_headline_lower_string = "🬂",
                },
            })
        end
    },
```

Then, I get this:

![image](https://github.com/user-attachments/assets/c72114f1-8846-4df0-9f79-e34796d2633b)

**Comment by jamespeapen** [comment]
I'll look into the headlines plugin - thanks!

**Comment by lokesh-krishna** [comment]
Sorry to hijack but was really excited that someone had the exact same issue as me and resolved it.  
However I don't have markdown highlighting disabled so might there be anything else I can take a look at to figure out why I still can't get those specific commands to work?

**Comment by jalvesaq** [comment]
There is an example of minimal configuration at the Wiki. If it works, then
you can gradually add your own configuration to discover what is breaking
it.

**Comment by lokesh-krishna** [comment]
Thanks, I tried it with just the minimal configuration but still get the "Not inside R or Python code chunk" message. At my wit's end here about just what could have changed since I do remember this working just fine for the longest time.

**Comment by jalvesaq** [comment]
I tried the _Almost Zero configuration_ and, indeed, it doesn't work (I will delete the subsection). However, the _Minimal configuration_ (using `lazy.nvim`) works.

**Comment by jalvesaq** [comment]
Trying again... It's not working. I'll fix the example.

**Comment by jalvesaq** [comment]
I updated the example in the Wiki. I had to enable the tree-sitter based highlighting:

```lua
        highlight = {
	  enable = true,
	},
```

**Comment by lokesh-krishna** [comment]
I simply can't thank you enough for your patience!
I built my configuration up from the minimal configuration and finally figured out exactly what it is I was missing. 

I use the `opts` spec to configure plugins and nvim-treesitter needs its main module specified with the `main` spec. Simply adding `main = "nvim-treesitter.configs` fixed it!

**Comment by jalvesaq** [comment]
Thank you for your feedback! It may help others.

**Comment by PMassicotte** [comment]
For my understanding, how enabling highlighting solved the issue?

**Comment by jalvesaq** [comment]
I've just tried: with my complete Neovim configuration even disabling tree-sitter highlighting the commands to send code to R still works. I guess that something must wake up the tree-sitter and, in the minimal configuration, that thing is the highlighting.

**Comment by jalvesaq** [comment]
While we don't discover what exactly is required to wake up the tree-sitter, I added tree-sitter-based highlighting as a pre-requisite: https://github.com/R-nvim/R.nvim/commit/a28189b8f3a3c2786d4bb749b8d9eeb7eafe8932

**Comment by conig** [comment]
I've just had this problem re-emerge with the default config. I fixed it by adding:

```
vim.treesitter.language.register('markdown', 'rmd')
```

to my `init.lua`

**Comment by jalvesaq** [comment]
@conig, the reemerged issue is going to be fixed by #239.
